plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/service/JavaLanguageExtensionServiceClient.java

```diff
@@ -295,6 +298,40 @@ private void subscribe() {
                 .onFailure(error -> reject.apply(ServiceUtil.getPromiseError(error))));
   }
 
+  /**
+   * Returns JDT LS java core options.
+   *
+   * @param filter list of the specific options to return. If list is empty then all available
+   *     options will be returned.
+   */
+  public Promise<JavaCoreOptions> getJavaCoreOptions(List<String> filter) {
+    return Promises.create(
+        (resolve, reject) ->
+            requestTransmitter
+                .newRequest()
+                .endpointId(WS_AGENT_JSON_RPC_ENDPOINT_ID)
+                .methodName(GET_JAVA_CORE_OPTIONS)
+                .paramsAsListOfString(filter)
+                .sendAndReceiveResultAsDto(JavaCoreOptions.class, REQUEST_TIMEOUT)
+                .onSuccess(resolve::apply)
+                .onTimeout(() -> onTimeout(reject))
+                .onFailure(error -> reject.apply(ServiceUtil.getPromiseError(error))));
+  }
+
+  public Promise<Boolean> updateJavaCoreOptions(JavaCoreOptions javaCoreOptions) {
```

Comment:
Please add a comment how it behaves (merge or replace existing settings. What happened if an empty map given, etc.)